src/test.js

```diff
@@ -113,24 +111,30 @@ function getNotStartedModules( startModule ) {
 
 Test.prototype = {
 	before: function() {
-		var module = this.module,
+		const module = this.module,
 			notStartedModules = getNotStartedModules( module );
 
 		// ensure the callbacks are executed serially for each module
-		var callbackPromises = notStartedModules.reduce( ( promiseChain, startModule ) => {
-			return promiseChain.then( () => {
+		const callbackPromises = notStartedModules.reduce( ( promiseChain, startModule ) => {
+			const moduleStartCallback = () => {
 				startModule.stats = { all: 0, bad: 0, started: now() };
 				emit( "suiteStart", startModule.suiteReport.start( true ) );
 				return runLoggingCallbacks( "moduleStart", {
 					name: startModule.name,
 					tests: startModule.tests
 				} );
-			} );
+			};
+
+			return promiseChain.then( moduleStartCallback, moduleStartCallback );
```

Comment:
Looks like this will swallow any errors at this point - is that the desired behavior here?

Reply:
@mikrostew I assuming you are talking about `const callbackPromises = notStartedModules.reduce(...);` ?

if any of the promise from callbackPromises reject.. the whole promise should reject.. which should be handled on line 154:
```
return callbackPromises.then( testStartHandler, ( err ) => {
		return Promise.reject( err ).catch( ( err ) => {
			setTimeout( testStartHandler );
			throw err;
		} );
	} );
```

Reply:
Read this again, this shouldn't be a problem, but I can understand why it looks like the error is swallowed.
For a bit more context.. `const callbackPromises = notStartedModules.reduce(...);` ensures the callbacks from notStartedModules are promisfied sequentually, so each callback that was made into a promise, an awaited for until the next callback is executed and made into a promise.
so You will get an chain of promises. In that chain, if there is one rejected promise, then the whole promise chain is rejected. 

`return promiseChain.then( moduleStartCallback, moduleStartCallback );` in this case, only ensures the next callback is promisfied even if the previous promise rejects. But as stated above, lien 154 will ensure the error is thrown.

I hope that makes more sense.. I think I should likely find a better way to code the explanation above up